src/com/itacademy/eshop/shop/Eshop.java

```diff
@@ -72,4 +73,14 @@ public Product findProductByName(String name) throws ProductNotFoundException {
         throw new ProductNotFoundException("Product with name " + name + " not found.");
 
     }
-}
+
+    public void duplicateProduct(String name ) throws DuplicateProductException {
+        for (Product product: products){
+            if (product.getName().equals(name)){
+                throw new DuplicateProductException("Product with name " + name + " is dublicate");
+
+        }
+    }
```

Comment:
formatting: the '}' should aligned with their respective lines statements: for and if